### PR TITLE
New Workspace RBAC changes

### PR DESCRIPTION
**Pull request by jwitz**
Resolves https://github.com/astronomer/docs/issues/2676
Resolves https://github.com/astronomer/docs/issues/2585

Release notes will arrive in a separate PR

Also tagging @RNHTTR for review to make sure there aren't any expected actions missing from the permissions list

### CODE REVIEW

File: astro/user-permissions.md [hunk 1]
```diff
@@ -41,30 +41,38 @@ To manage users in an Organization, see [Manage Organization users](manage-organ
...
+| Create, update, and delete Astro alerts                                     |                      | ✔️                    | ✔️                      | ✔️                   |
+| View Airflow connections, variables, plugins, providers, pools, and XComs   |                      | ✔️                    | ✔️                      | ✔️                   |
+| Update Airflow connections, variables, plugins, providers, pools, and XComs |                      |                      | ✔️                      | ✔️                   |
+| Update Deployment configurations                                            |                      |                      | ✔️                      | ✔️                   |
+| Create and Delete Deployments                                               |                      |                      | ✔️                      | ✔️                   |
+| Create, Update and Delete Environment Variables                             |                      |                      | ✔️                      | ✔️                   |
```
Comment (chomipi88) [inline astro/user-permissions.md:71]:
```suggestion
| Create, Update and Delete Deployment Environment Variables                             |                      |                      | ✔️                      | ✔️                   |
```

**Comment by chomipi88** [comment]
The blog includes a table that is a bit more specific than the table in our docs. Can we just use the blog table?

https://www.astronomer.io/blog/introducing-updated-astro-workspace-roles-for-more-granular-permissions/

**Comment by RNHTTR** [comment]
I think we should cover everything in the top bar of the UI that not every user has access to:

image

Assuming Members can't interact with them, we're missing cluster activity and datasets.

> View Airflow connections, variables, plugins, providers, pools, and XComs

Who can create/update/delete these?

File: astro/user-permissions.md [hunk 2]
```diff
@@ -41,30 +41,38 @@ To manage users in an Organization, see [Manage Organization users](manage-organ
...
+| Create, update, and delete Astro alerts                                     |                      | ✔️                    | ✔️                      | ✔️                   |
+| View Airflow connections, variables, plugins, providers, pools, and XComs   |                      | ✔️                    | ✔️                      | ✔️                   |
+| Update Airflow connections, variables, plugins, providers, pools, and XComs |                      |                      | ✔️                      | ✔️                   |
+| Update Deployment configurations                                            |                      |                      | ✔️                      | ✔️                   |
+| Create and Delete Deployments                                               |                      |                      | ✔️                      | ✔️                   |
+| Create, Update and Delete Deployment Environment Variables                             |                      |                      | ✔️                      | ✔️                   |
```
Comment (jwitz) [inline astro/user-permissions.md:71]:
```suggestion
| Create, Update and Delete Deployment Environment Variables                  |                      |                      | ✔️                      | ✔️                   |
```

File: astro/user-permissions.md [hunk 3]
```diff
@@ -41,30 +41,38 @@ To manage users in an Organization, see [Manage Organization users](manage-organ
...
+
+The following table lists the specific permissions that each Workspace role has:
+
+| Permission                                                                  | **Workspace Member** | **Workspace Author** | **Workspace Operator** | **Workspace Owner** |
+| --------------------------------------------------------------------------- | -------------------- | -------------------- | ---------------------- | ------------------- |
+| View Workspace users                                                        | ✔️                    | ✔️                    | ✔️                      | ✔️                   |
+| View all Deployments in the Cloud UI                                        | ✔️                    | ✔️                    | ✔️                      | ✔️                   |
+| View DAGs in the Airflow UI                                                 | ✔️                    | ✔️                    | ✔️                      | ✔️                   |
+| View Airflow task logs                                                      | ✔️                    | ✔️                    | ✔️                      | ✔️                   |
```
Comment (jwitz) [inline astro/user-permissions.md:58]:
```suggestion
| View Airflow task logs                                                      | ✔️                    | ✔️                    | ✔️                      | ✔️                   |
| View Airflow datasets                                                       | ✔️                    | ✔️                    | ✔️                      | ✔️                   |
```

**Comment by jwitz** [comment]
@chomipi88 Do we know the scope of permissions for the "Cluster Activity" tab in 2.7? https://airflow.apache.org/blog/airflow-2.7.0/#cluster-activity-ui

**Comment by chomipi88** [comment]
The diagram also needs to be updated with new role names.
https://docs.astronomer.io/astro/astro-architecture#access-control-architecture

**Comment by chomipi88** [comment]
> @chomipi88 Do we know the scope of permissions for the "Cluster Activity" tab in 2.7? https://airflow.apache.org/blog/airflow-2.7.0/#cluster-activity-ui

@jwitz Just tested and only Workspace Owner would see the new Cluster Activity tab. If we want lower Workspace roles to see this new tab, their will be some work to support this in casbin and authz.

**Comment by chomipi88** [comment]
New Workspace RBAC will be released tomorrow along with a new CLI release to support the new role. Will there be a separate PR for Astro and CLI release notes?

File: astro/user-permissions.md [hunk 4]
```diff
@@ -41,30 +41,39 @@ To manage users in an Organization, see [Manage Organization users](manage-organ
...
-| Update Airflow connections and Variables            |                      |                      | ✔️                   |
-| Create, Update, and Delete API Keys                 |                      |                      | ✔️                   |
-| Update user roles and permissions                   |                      |                      | ✔️                   |
-| Invite users to a Workspace                         |                      |                      | ✔️                   |
-| Assign Teams to or remove from Workspaces           |                      |                      | ✔️                   |
-| Create, update and delete Workspace API tokens      |                      |                      | ✔️                   |
+A Workspace role grants a user or API token some level of access to a specific Workspace. If a user or API token has some level of access to a Workspace, that access applies to all Deployments in the Workspace. 
+
+- A **Workspace Member** has the least permissions in a Workspace and can only view the most basic details about Deployments and DAGs. Give a user this role if X
```
Comment (lzdanski) [inline astro/user-permissions.md:46]: Looks like some stand-in text didn't get fully updated. Took a stab at it. 
```suggestion
- A **Workspace Member** has the least permissions in a Workspace and can only view the most basic details about Deployments and DAGs. Give a user this role if they need to be able to view a DAG's run status or Deployment health.
```

**Review by lzdanski — COMMENTED** [review]
looks like one spot needs some added detail

File: astro/user-permissions.md [hunk 5]
```diff
@@ -41,30 +41,39 @@ To manage users in an Organization, see [Manage Organization users](manage-organ
...
+| Create, update, and delete Astro alerts                                     |                      | ✔️                    | ✔️                      | ✔️                   |
+| View Airflow connections, variables, plugins, providers, pools, and XComs   |                      | ✔️                    | ✔️                      | ✔️                   |
+| Update Airflow connections, variables, plugins, providers, pools, and XComs |                      |                      | ✔️                      | ✔️                   |
+| Update Deployment configurations                                            |                      |                      | ✔️                      | ✔️                   |
+| Create and Delete Deployments                                               |                      |                      | ✔️                      | ✔️                   |
+| Create, Update and Delete Deployment environment variables                  |                      |                      | ✔️                      | ✔️                   |
```
Comment (jwitz) [inline astro/user-permissions.md:72]:
```suggestion
| Create, Update and Delete Deployment environment variables                  |                      |                      | ✔️                      | ✔️                   |
| View the **Cluster Activity** tab in the Airflow UI                  |                      |                      | ✔️                      | ✔️                   |
```

File: astro/user-permissions.md [hunk 6]
```diff
@@ -41,30 +41,40 @@ To manage users in an Organization, see [Manage Organization users](manage-organ
...
+| View the **Cluster Activity** tab in the Airflow UI                  |                      |                      | ✔️                      | ✔️                   |
```
Comment (jwitz) [inline astro/user-permissions.md:73]:
```suggestion
| View the **Cluster Activity** tab in the Airflow UI                       |                      |                      | ✔️                      | ✔️                   |
```

File: astro/user-permissions.md [hunk 7]
```diff
@@ -41,30 +41,40 @@ To manage users in an Organization, see [Manage Organization users](manage-organ
...
+| View Airflow connections, variables, plugins, providers, pools, and XComs   |                      | ✔️                    | ✔️                      | ✔️                   |
+| Update Airflow connections, variables, plugins, providers, pools, and XComs |                      |                      | ✔️                      | ✔️                   |
+| Update Deployment configurations                                            |                      |                      | ✔️                      | ✔️                   |
+| Create and Delete Deployments                                               |                      |                      | ✔️                      | ✔️                   |
+| Create, Update and Delete Deployment environment variables                  |                      |                      | ✔️                      | ✔️                   |
+| View the **Cluster Activity** tab in the Airflow UI                       |                      |                      | ✔️                      | ✔️                   |
```
Comment (jwitz) [inline astro/user-permissions.md:73]:
```suggestion
| View the **Cluster Activity** tab in the Airflow UI                         |                      |                      | ✔️                      | ✔️                   |
```

File: astro/user-permissions.md [hunk 8]
```diff
@@ -41,30 +41,40 @@ To manage users in an Organization, see [Manage Organization users](manage-organ
...
+| View Airflow connections, variables, plugins, providers, pools, and XComs   |                      | ✔️                    | ✔️                      | ✔️                   |
+| Update Airflow connections, variables, plugins, providers, pools, and XComs |                      |                      | ✔️                      | ✔️                   |
+| Update Deployment configurations                                            |                      |                      | ✔️                      | ✔️                   |
+| Create and Delete Deployments                                               |                      |                      | ✔️                      | ✔️                   |
+| Create, Update and Delete Deployment environment variables                  |                      |                      | ✔️                      | ✔️                   |
+| View the **Cluster Activity** tab in the Airflow UI                         |                      |                      | ✔️                      | ✔️                   |
```
Comment (jwitz) [inline astro/user-permissions.md:73]:
```suggestion
| View the **Cluster Activity** tab in the Airflow UI                         |                      |                      |                        | ✔️                   |
```

**Comment by jwitz** [comment]
@chomipi88 Release notes are now in https://github.com/astronomer/docs/pull/2758

**Review by chomipi88 — APPROVED** [review]

**Review by lzdanski — APPROVED** [review]
Checked on Preview and changes look good/links are working. It also looks like you've had a technical review approval, but you might still need a stamp from @RNHTTR for changes requested.

**Pull request merged**